Feature: shutter animation

andi34 opened the pull request:
<!--
    Thank you for contributing!
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)
<!-- Example:
- [x] Bug fix
-->
- [x] New feature

<!--
    Please ensure your pull request is ready:

    - Please run 'yarn build' before submitting to have consistent formatting for both JavaScript & SCSS
    - Please run 'yarn eslint' to make sure there's no lint issues
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Add a simple shutter animation while taking a picture. Implementation from @up-87,
fixed animation on retry / next collage image.
Made shutter animation an option inside Adminpanel, animation is enabled by default.

#### Is there anything you'd like reviewers to focus on?

up-87 committed:
ui: make taking pictures more beautiful

Currently just two layers displayed over everything - not really optimal.
Doesn't work on all pictures of a collage yet.

Change-Id: I756a2d46cf779d0af78b19c7c90473ebc391070a

andi34 committed:
core: fix animation on next image (retry / next collage image)

- reset aperture and blocker css values to default

Change-Id: I32f065c7d03bb28c86478f1915c9d08859008010

andi34 committed:
core: allow to disable the shutter animation

Change-Id: I77c3cd93ed1136ba6dc004083889b64d6bd301f0

andi34 commented on src/js/core.js:
```diff
@@ -341,6 +341,7 @@ const photoBooth = (function () {
 
     api.cheese = function (photoStyle, retry = 0) {
+        api.stopPreviewVideo(CameraDisplayMode.COUNTDOWN);
         photoboothTools.console.logDev('Photostyle: ' + photoStyle);
```
?

up-87 replied:
As I said I tried a bit around with it and tried to keep the cheese (Wouldn’t make any sense though. The photo was taken at the time of the fade to white) That‘s why I considered the branch dirty. This line can be deleted.

up-87 commented on src/js/core.js:
```diff
@@ -426,6 +427,10 @@ const photoBooth = (function () {
     };
 
     api.callTakePicApi = function (data, retry = 0) {
+        const blocker = $('#blocker');
```
Should be moved up to the other jquery constants.

up-87 commented on src/js/core.js:
```diff
@@ -434,6 +439,27 @@ const photoBooth = (function () {
                 totalTime = endTime - startTime;
                 photoboothTools.console.log('took ' + data.style, result);
                 photoboothTools.console.logDev('Taking picture took ' + totalTime + 'ms');
+                const aperture = $('#aperture');
```
Should be moved up to the other jquery constants.

andi34 committed:
core: add own animation functions

Move the shutter start and stop animation to it's own function,
this way we are more flexible while adjusting things if needed.

Change-Id: If4b9c8c4d3f664bb44c40b5a19124b86bac52305

andi34 committed:
livechroma: use shutter animation

Change-Id: I5750bfd63f533f24a755cef2788e2ac926534595

andi34 commented:
@up-87 anything left? IMO can be rebased and merged if there's currently no further improvement.
